run_tests.sh

```diff
@@ -0,0 +1,7 @@
st,RDFDatetimeTest,FilehashTest,XoredSearchingTest,GRRFuseTest,ClientCommsTest,FilterTests,AFF4SparseImageTest,ByteSizeTest,OSXEnumerateRunningServicesTest,TimelineTest,TestTimelines,TestPackedVersionedCollection,VFSDirectoryTest,FindTest,RekallTests,RDFIntegerTest,TestExecutePython,MethodTest,EmailPluginTest,PackedVersionedCollectionCompactorTest,OSXMemoryTest,HashFileStoreTest,ArtifactFlowTest,AFF4ObjectLabelsListTest,TestExports,AuthTest,ConfigLibTest,GrepTests,GrrKbTest,TestArtifactCollectors,TriggersTest,UsersTest,SshdConfigTest,TestLinuxMemory,DurationTest,ClientURNTests,SqliteDataStoreTest,DataStoreTest,AFF4ObjectLabelTest,TestCheckFlows,TDBDataStoreBenchmarks,ThreadPoolTest,AFF4GRRTest,ChromeHistoryTest,GetClientStatsActionTest,TestSearchFileContent,TestRegistryFinderFlow,AFF4LabelsIndexTest,ListProcessesTest,Ipv6UtilsTest,RDFStructsTest,BuildTests,TestPlistFlows,FileStoreHashTest,ApiConfigRendererTest,IEHistoryTest,RunningStatsTest,AdministrativeFlowTests,YamlConfigTest,RDFValueBaseTest,RDFBytesTest,FileExportPluginTest,TestCopyPathToFile,SqliteDataStoreCSVBenchmarks,RDFValueBenchmark,TDBDataStoreTest,TestLinuxMemoryAnalysis,StoreTests,DataStoreBenchmarks,HintsTests,RekallResponseConverterTest,WindowsRegistryParserTest,ClientUtilsTest,GRRTempFileTestDirectory,EmptyActionTest,FakeDataStoreBenchmarks,AverageMicroBenchmarks,TestListVolumeShadowCopies,WindowsPersistenceMechanismsParserTest,HashDigestTest,WMIParserTest,CollectionExportPluginTest,StatsTests,MasterTest,ApiAff4RendererTest,SendEmailTests,TestClientInterrogate,TestExportHuntResultsFilesAsArchive,RDFURNTest,LinuxSoftwareParserTest,ArtifactParserTest,TestFilesystem,HuntTest,BatchConverterTest,TestArtifactCollectorsInteractions,OSXQuarantineTest,CollectorTest,CollectionFilesExportPluginTest,StandardHuntTest,UserTests,TestSearchFileContentWithFixture,TestCollections,ArtifactParserTests,DarwinPersistenceMechanismsParserTest,RDFValueCollectionApiObjectRendererTest,WebHistoryFlowTest,HTTPDataStoreTest,EmbeddedRDFValueTest,DataAgnosticExportConverterTest,GRRSeleniumTest,SqliteDataStoreBenchmarks,TestSystemRootSystemDriveFallbackFlow,OutputpluginsTest,ConfigFileTest,FlowTest,RegistryFlowTest,ExportTest,MatchMethodTests,ResultSetTest,FakeDataStoreTest,HintDefinitionTests,TestAdministrativeFlows,GRRTempFileTestFilename,FilestoreStatsCronFlowTest,ForemanTests,ReportsTest,GRRBaseTest,TestEndToEndTestFlow,DictTest,TestArtifactCollectorsRealArtifacts,KnowledgeBaseUserMergeTest,HandlerTests,NetstatTest,BasicContextTests,ArtifactHandlingTest,ObjcTest,TestWindowsMemory,WindowsActionTests,GrepTest,SystemCronFlowTest,UtilsTest,DataStoreCSVBenchmarks,CheckLoaderTests,TDBDataStoreCSVBenchmarks,ArtifactKBTest,MultiShardedQueueManagerTest,ConfigActionTest,SearchTest,RegistryVFSTests,OSXDriverTests,RekallTestSuite,GeneralFlowsTest,FlowStateTest,TestCryptoTypeInfos,AFF4Benchmark,GRRArtifactTest,GRRFuseDatastoreOnlyTest,TestOSXFileParsing,AFF4SymlinkTest,FlowTestsBaseclass,ConditionTest,RekallVADParserTest
```

Comment:
if you expect these tests to fail, then let them fail - travis will tell
you they are failing and you can fix it then. disabling failed tests seem
like cheating :-). I wonder why they are failing because they do pass in a
regular source checkout. It might be something simple due to the way you
are installing the travil environ?

On 9 February 2015 at 16:25, Dionysis Zindros <notifications@github.com>
wrote:

> In run_tests.sh
> <https://github.com/google/grr/pull/113#discussion_r24335171>:
>
> > @@ -0,0 +1,7 @@
> > +#!/bin/bash
> > +cd ..
> > +echo 'Running tests...'
> > +EXCLUDE_TESTS=TestTransfer,FlowLimitTests,TestFileFinderFlow,ListVADBinariesTest,ApiRenderersTest,RDFDatetimeSecondsTest,GrrWorkerTest,RDFValueArrayTest,VFSGRRClientApiObjectRendererTest,DeleteGRRTempFiles,GlobExpressionTest,CheckAccessHelperTest,ArtifactTest,BasicFlowTest,CSVOutputPluginTest,TestMemoryAnalysis,PathSpecTest,IndexTest,TestApplyLabelsToClientsFlow,TestMemoryCollector,OSXFilesystemTests,ChecksTestBase,HashFileStoreExportPluginTest,TestWebHistoryWithArtifacts,ApiHuntsListRendererTest,QueueManagerTest,TestFingerprintFlow,TypeInfoTest,PlistTest,OSSpecificClientTests,FlowCreationTest,GRRFuseTestBase,NetworkAddressTests,RDFStringTest,HTTPClientTests,MemoryTest,LinuxFileParserTest,GRRFEServerTest,TestDebugFlows,TestNetworkByteLimits,OSXVersionTests,ProbeTest,AFF4ObjectTest,TestFindFlow,HTTPDataStoreCSVBenchmarks,SQLiteFileTest,ApiAff4IndexRendererTest,CheckResultsTest,AFF4IndexSetTest,HTTPDataStoreBenchmarks,ArtifactTests,TestRegistryFlows,StatsStoreTest,MicroBe
>  nchmarks
>  ,TestWebHistory,BlobImageTest,TestInterpolatePath,GRRUserTest,EntryPointTest,TestAuditSystem,LexerTests,ProfileServerTest,OSXClientTests,SignedBlobTest,AccessControlTest,BuildConfigTests,LinuxReleaseParserTest,CheckRegistryTests,StatsStoreDataQueryTest,FileStoreTest,AFF4ObjectApiObjectRendererTest,OSXLaunchdJobDictTest,ActionTest,VFSTest,Firefox3HistoryTest,CheckTest,GenericRDFProtoTest,CronTest,RDFDatetimeTest,FilehashTest,XoredSearchingTest,GRRFuseTest,ClientCommsTest,FilterTests,AFF4SparseImageTest,ByteSizeTest,OSXEnumerateRunningServicesTest,TimelineTest,TestTimelines,TestPackedVersionedCollection,VFSDirectoryTest,FindTest,RekallTests,RDFIntegerTest,TestExecutePython,MethodTest,EmailPluginTest,PackedVersionedCollectionCompactorTest,OSXMemoryTest,HashFileStoreTest,ArtifactFlowTest,AFF4ObjectLabelsListTest,TestExports,AuthTest,ConfigLibTest,GrepTests,GrrKbTest,TestArtifactCollectors,TriggersTest,UsersTest,SshdConfigTest,TestLinuxMemory,DurationTest,ClientURNTests,SqliteDat
>  aStoreTe
>  st,DataStoreTest,AFF4ObjectLabelTest,TestCheckFlows,TDBDataStoreBenchmarks,ThreadPoolTest,AFF4GRRTest,ChromeHistoryTest,GetClientStatsActionTest,TestSearchFileContent,TestRegistryFinderFlow,AFF4LabelsIndexTest,ListProcessesTest,Ipv6UtilsTest,RDFStructsTest,BuildTests,TestPlistFlows,FileStoreHashTest,ApiConfigRendererTest,IEHistoryTest,RunningStatsTest,AdministrativeFlowTests,YamlConfigTest,RDFValueBaseTest,RDFBytesTest,FileExportPluginTest,TestCopyPathToFile,SqliteDataStoreCSVBenchmarks,RDFValueBenchmark,TDBDataStoreTest,TestLinuxMemoryAnalysis,StoreTests,DataStoreBenchmarks,HintsTests,RekallResponseConverterTest,WindowsRegistryParserTest,ClientUtilsTest,GRRTempFileTestDirectory,EmptyActionTest,FakeDataStoreBenchmarks,AverageMicroBenchmarks,TestListVolumeShadowCopies,WindowsPersistenceMechanismsParserTest,HashDigestTest,WMIParserTest,CollectionExportPluginTest,StatsTests,MasterTest,ApiAff4RendererTest,SendEmailTests,TestClientInterrogate,TestExportHuntResultsFilesAsArchive,R
>  DFURNTes
>  t,LinuxSoftwareParserTest,ArtifactParserTest,TestFilesystem,HuntTest,BatchConverterTest,TestArtifactCollectorsInteractions,OSXQuarantineTest,CollectorTest,CollectionFilesExportPluginTest,StandardHuntTest,UserTests,TestSearchFileContentWithFixture,TestCollections,ArtifactParserTests,DarwinPersistenceMechanismsParserTest,RDFValueCollectionApiObjectRendererTest,WebHistoryFlowTest,HTTPDataStoreTest,EmbeddedRDFValueTest,DataAgnosticExportConverterTest,GRRSeleniumTest,SqliteDataStoreBenchmarks,TestSystemRootSystemDriveFallbackFlow,OutputpluginsTest,ConfigFileTest,FlowTest,RegistryFlowTest,ExportTest,MatchMethodTests,ResultSetTest,FakeDataStoreTest,HintDefinitionTests,TestAdministrativeFlows,GRRTempFileTestFilename,FilestoreStatsCronFlowTest,ForemanTests,ReportsTest,GRRBaseTest,TestEndToEndTestFlow,DictTest,TestArtifactCollectorsRealArtifacts,KnowledgeBaseUserMergeTest,HandlerTests,NetstatTest,BasicContextTests,ArtifactHandlingTest,ObjcTest,TestWindowsMemory,WindowsActionTests,Grep
>  Test,Sys
>  temCronFlowTest,UtilsTest,DataStoreCSVBenchmarks,CheckLoaderTests,TDBDataStoreCSVBenchmarks,ArtifactKBTest,MultiShardedQueueManagerTest,ConfigActionTest,SearchTest,RegistryVFSTests,OSXDriverTests,RekallTestSuite,GeneralFlowsTest,FlowStateTest,TestCryptoTypeInfos,AFF4Benchmark,GRRArtifactTest,GRRFuseDatastoreOnlyTest,TestOSXFileParsing,AFF4SymlinkTest,FlowTestsBaseclass,ConditionTest,RekallVADParserTest
>
> I'm excluding these because they fail and I plan to fix them one-by-one.
> However, fixing them one-by-one takes a long time and this pull request is
> starting to diverge from master in a way that makes future conflict
> resolution painful.
>
> We'd like to have a working travis build ready and merged soon, as it will
> help with the C++ build process, so this is a temporary solution for
> expediting this merge. That's why they're being excluded.
>
> I see this as a positive first step towards getting a proper travis build
> going: It's better to have only a handful of tests running instead of
> nothing. So I think this PR is worth merging as it is. This strategy is
> followed by other GitHub projects gradually enabling travis as well.
>
> As for including the ones we want instead of excluding, this is a
> situation of a blacklist VS whitelist. These tests are explicitly black
> listed as "potentially failing" and must be looked at. However, I don't
> want to use a white list, as from now on we should use travis as an
> indicator of a successful build. Therefore, if another pull request comes
> in and includes a new test, it should be included by default.
>
> That way, we will be able to validate the build and verify the correctness
> of patches by open source contributors and start being more open and
> cooperating with external parties in the development of Grr.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/grr/pull/113/files#r24335171>.
>